autopts/ptsprojects/stack/layers/pbp.py

```diff
@@ -31,3 +33,11 @@ def wait_public_broadcast_event_found_ev(self, addr_type, addr, broadcast_name,
             self.event_queues[defs.BTP_PBP_EV_PUBLIC_BROADCAST_ANNOUNCEMENT_FOUND],
             lambda ev: (addr_type, addr, broadcast_name) == (ev['addr_type'], ev['addr'], ev['broadcast_name']),
             timeout, remove)
+
+    def set_program_info(self, program_info):
+        if isinstance(program_info, str) and len(program_info > 0):
+            self.program_info = program_info
+
+    def set_broadcast_name(self, broadcast_name):
+        if isinstance(broadcast_name, str) and len(broadcast_name > 0):
+            self.broadcast_name = broadcast_name
```

Comment:
Should they raise an exception instead of failing silently?

Reply:
Please fail loudly, it makes debugging / adding support for other stacks / ptsprojects much easier.

Reply:
Done